games-strategy/freeciv/freeciv-3.0.9.ebuild

```diff
@@ -0,0 +1,185 @@
+	sys-libs/zlib
+	auth? (
+		!mysql? ( ( !sqlite? ( dev-db/mysql-connector-c:= ) ) )
+		mysql? ( dev-db/mysql-connector-c:= )
+		sqlite? ( dev-db/sqlite:3 )
+	)
+	dedicated? ( aimodules? ( dev-libs/libltdl ) )
+	!dedicated? (
+		media-libs/libpng
+		gtk? ( x11-libs/gtk+:3 )
+		mapimg? ( media-gfx/imagemagick:= )
+		modpack? ( x11-libs/gtk+:3 )
+		nls? ( virtual/libintl )
+		qt5? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qtwidgets:5
+		)
+		!sdl? ( !gtk? ( x11-libs/gtk+:3 ) )
+		sdl? (
+			media-libs/libsdl2[video]
+			media-libs/sdl2-gfx
+			media-libs/sdl2-image[png]
+			media-libs/sdl2-ttf
+		)
+		server? ( aimodules? ( sys-devel/libtool ) )
+		sound? (
+			media-libs/libsdl2[sound]
+			media-libs/sdl2-mixer[vorbis]
+		)
+	)
+	readline? ( sys-libs/readline:= )
+	system-lua? ( ${LUA_DEPS} )
+"
+DEPEND="${RDEPEND}
+	!dedicated? ( x11-base/xorg-proto )
+"
+# Calls gzip during build
+BDEPEND="
+	app-arch/gzip
+	virtual/pkgconfig
+	!dedicated? ( nls? ( sys-devel/gettext ) )
+"
+
+S="${WORKDIR}/${MY_P}"
+
+pkg_setup() {
+	if ! use dedicated && ! use server ; then
+		ewarn "Disabling server USE flag will make it impossible to start local"
+		ewarn "games, but you will still be able to join multiplayer games."
+	fi
+
+	use system-lua && lua-single_pkg_setup
+}
+
+src_configure() {
+	local myclient=() mydatabase=() myeconfargs=()
+
+	if use auth ; then
+		if ! use mysql && ! use sqlite ; then
+			einfo "No database backend chosen, defaulting"
+			einfo "to mysql!"
+			mydatabase=( mysql )
+		else
+			use mysql && mydatabase+=( mysql )
+			use sqlite && mydatabase+=( sqlite3 )
+		fi
+	else
+		mydatabase=( no )
+	fi
+
+	if use dedicated ; then
+		myclient=( no )
+		myeconfargs+=(
+			--enable-server
+			--enable-freeciv-manual=html
+		)
+	else
+		if use !sdl && use !gtk && ! use qt5 ; then
+			einfo "No client backend given, defaulting to gtk3 client!"
+			myclient=( gtk3 )
+		else
+			use sdl && myclient+=( sdl2 )
+			# Since all gtk3 in gentoo is >= 3.22 we can use the better client
+			use gtk && myclient+=( gtk3.22 )
+			if use qt5 ; then
+				local -x MOCCMD=$(qt5_get_bindir)/moc
+				myclient+=( qt )
+			fi
+		fi
+		myeconfargs+=(
+			$(use_enable server)
+			$(use_enable server freeciv-manual html )
+		)
+	fi
+
+	myeconfargs+=(
+		--enable-aimodules="$(usex aimodules "yes" "no")"
+		--enable-client="${myclient[*]}"
+		--enable-fcdb="${mydatabase[*]}"
+		--enable-fcmp="$(usex modpack "gtk3" "no")"
+		--enable-ipv6
+		# disabling shared libs will break aimodules USE flag
+		--enable-shared
+		--localedir=/usr/share/locale
+		--with-appdatadir="${EPREFIX}"/usr/share/metainfo
+		$(use_enable mapimg)
+		$(use_enable nls)
+		$(use_enable sound sdl-mixer)
+		$(use_enable system-lua sys-lua)
+		$(use_with readline)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	default
+
+	if use dedicated ; then
+		rm -rf "${ED}"/usr/share/pixmaps || die
+		rm -f "${ED}"/usr/share/man/man6/freeciv-{client,gtk2,gtk3,modpack,qt,sdl,xaw}* || die
```

Comment:
These files should be predictable, especially when put behind use-conditional. We generally dislike `rm -f`, as they never get removed and should only be used for unpredictable files.